guides/source/active_model_basics.md

```diff
@@ -44,76 +67,331 @@ class EmailContact
 end
 ```
 
-When including `ActiveModel::API` you get some features like:
+```irb
+irb> email_contact = EmailContact.new(name: "David", email: "david@example.com", message: "Hello World")
+
+irb> email_contact.name # attribute assignment
+=> "David"
+
+irb> email_contact.to_model == email_contact # conversion
+=> true
+
+irb> email_contact.model_name.name # naming
+=> EmailContact
+
+irb> email_contact.human_attribute_name("name") # translation if the locale is set
+=> "Name"
+
+irb> email_contact.valid? # validations
+=> true
+```
+
+Any class that includes `ActiveModel::API` can be used with `form_with`,
+`render` and any other [Action View helper
+methods](https://api.rubyonrails.org/classes/ActionView/Helpers.html), just like
+Active Record objects.
+
+For example, `form_with` can be used to create a form for an `EmailContact`
+object as follows:
+
+```html+erb
+<%= form_with model: EmailContact.new do |form| %>
+  <%= form.text_field :name %>
+<% end %>
+
+# =>
+<form action="/email_contacts" method="post" data-remote="true">
+  <input type="text" name="email_contact[name]">
+</form>
+```
+
+`render` can be used to render a partial with the object as a local variable:
+
+```html+erb
+<%= render partial: "email_contact", email_contact: EmailContact.new(name: "David", email: "david@example.com", message: "Hello World") %>
+```
+
+### Attributes
 
-- model name introspection
-- conversions
-- translations
-- validations
+`ActiveModel::Attributes` allows you to define data types, set default values,
+and handle casting and serialization on plain Ruby objects. This can be useful
+for form data which will produce Active Record-like conversion for things like
+dates and booleans on regular objects.
 
-It also gives you the ability to initialize an object with a hash of attributes,
-much like any Active Record object.
+To use Attributes, include the module in your model class and define your
+attributes using the `attribute` macro. It accepts a name, a cast type, a
+default value, and any other options supported by the attribute type.
+
+```ruby
+class Person
+  include ActiveModel::Attributes
+
+  attribute :name, :string
+  attribute :date_of_birth, :date
+  attribute :active, :boolean, default: true
+end
+```
 
 ```irb
-irb> email_contact = EmailContact.new(name: 'David', email: 'david@example.com', message: 'Hello World')
-irb> email_contact.name
-=> "David"
-irb> email_contact.email
-=> "david@example.com"
-irb> email_contact.valid?
+irb> person = Person.new
+
+irb> person.name = "Jane"
+irb> person.name
+=> "Jane"
+
+# casts the string to a date set by the attribute
+irb> person.date_of_birth = "2020-01-01"
+irb> person.date_of_birth
+=> Wed, 01 Jan 2020
+irb> person.date_of_birth.class
+=> Date
+
+# uses the default value set by the attribute
+irb> person.active
 => true
-irb> email_contact.persisted?
+
+# casts the string to a boolean set by the attribute
+irb> person.active = "0"
```

Comment:
Is there a reason why it has to be a string? If not, using `0` might be cleaner for the example.

```
>> person.active = 0
=> 0
>> person.active
=> false
>> person.active = 1
=> 1
>> person.active
=> true
```

```suggestion
irb> person.active = 0
```